### PR TITLE
Use a wrapper script to locate kubefed and kubectl binaries instead of directly constructing their paths.

### PR DESCRIPTION
**Pull request by madhusudancs**
This fixes the e2e failures that is now switched to using kubefed.

cc @kubernetes/sig-federation-pr-reviews

### CODE REVIEW

File: federation/develop/kubefed.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+elif [[ ! -x "${KUBEFED_PATH}" ]]; then
+  {
+    echo "KUBEFED_PATH environment variable set to '${KUBEFED_PATH}', but "
+    echo "this doesn't seem to be a valid executable."
+  } >&2
+  exit 1
+fi
+kubefed="${KUBEFED_PATH:-${kubefed}}"
+
+"${kubefed}" "${@+$@}"
```
Comment (nikhiljindal) [inline federation/develop/kubefed.sh]: what does `@+$@` do? Maybe add a comment?

File: federation/cluster/federation-up.sh [hunk 2]
```diff
@@ -73,13 +69,17 @@ function create_cluster_secrets() {
     name=$(echo "${dir}" | sed -e "s/_/-/g")  # Replace "_" by "-"
     name=${name:0:252}
     kube::log::status "Creating secret with name: ${name} in namespace ${FEDERATION_NAMESPACE}"
-    ${kubectl} create secret generic ${name} --from-file="${base_dir}/${dir}/kubeconfig" --namespace="${FEDERATION_NAMESPACE}"
+    "${KUBE_ROOT}/cluster/kubectl.sh" create secret generic ${name} --from-file="${base_dir}/${dir}/kubeconfig" --namespace="${FEDERATION_NAMESPACE}"
```
Comment (nikhiljindal) [inline federation/cluster/federation-up.sh]: Use the same `get_bin` for kubectl as well?

**Comment by nikhiljindal** [comment]
Thanks, LG with a 2  minor comments.
Feel free to self LGTM after fixing those

File: federation/develop/kubefed.sh (hunk 1 above)
Comment (madhusudancs) [inline federation/develop/kubefed.sh]: It means, use `$@` if it is set, null string otherwise.

File: federation/cluster/federation-up.sh (hunk 2 above)
Comment (madhusudancs) [inline federation/cluster/federation-up.sh]: I did not understand this comment. Can you elaborate?

**Review by madhusudancs — COMMENTED** [review]
@nikhiljindal PTAL.

**Comment by madhusudancs** [comment]
/lgtm
/approve

(self-lgtm'ing and approving based on @nikhiljindal's comment).

File: federation/cluster/federation-up.sh (hunk 2 above)
Comment (nikhiljindal) [inline federation/cluster/federation-up.sh]: oh ok I got it wrong last time. What you have LG.

**Comment by nikhiljindal** [comment]
lgtm, thanks

**Comment by nikhiljindal** [comment]
.gtm

**Pull request merged**